jdaviz/configs/imviz/tests/test_orientation.py

```diff
@@ -1,31 +1,146 @@
+            # same on both entries.
+            # 0.25 offset probably introduced by fake WCS layer.
+            assert_allclose(mp._obj.marks["imviz-0"].x, -0.25)
+            assert_allclose(mp._obj.marks["imviz-0"].y, -0.25)
+
+            mp.clear_table()
+
+    def test_markers_plugin_recompute_positions_wcs_to_pixels(self):
+        lc_plugin = self.imviz.plugins['Orientation']
+        lc_plugin.link_type = 'WCS'
+
+        # Blink to second image, if we have to.
+        if self.viewer.top_visible_data_label != "has_wcs_2[SCI,1]":
+            self.viewer.blink_once()
+
+        label_mouseover = self.imviz.app.session.application._tools['g-coords-info']
+        mp = self.imviz.plugins['Markers']
+
+        with mp.as_active():
+            # (0, 0) on second image, but linked by WCS.
+            label_mouseover._viewer_mouse_event(
+                self.viewer, {'event': 'mousemove', 'domain': {'x': 0, 'y': 0}})
+            mp._obj._on_viewer_key_event(self.viewer, {'event': 'keydown', 'key': 'm'})
+
+            # (0, 0) on first image.
+            self.viewer.blink_once()
+            label_mouseover._viewer_mouse_event(
+                self.viewer, {'event': 'mousemove', 'domain': {'x': 0, 'y': 0}})
+            mp._obj._on_viewer_key_event(self.viewer, {'event': 'keydown', 'key': 'm'})
+
+            lc_plugin.link_type = 'Pixels'
+
+            # Both marks now get separated, so this means X and Y w.r.t. reference
+            # are different on both entries.
+            # 0.25 offset probably introduced by fake WCS layer.
+            assert_allclose(mp._obj.marks["imviz-0"].x, [1.25, 0.25])
+            assert_allclose(mp._obj.marks["imviz-0"].y, 0.25)
+
+            mp.clear_table()
+
+
+class TestNonDefaultOrientation(BaseImviz_WCS_WCS):
+    def test_N_up_multi_viewer(self):
+        lc_plugin = self.imviz.plugins['Orientation']
+        lc_plugin.link_type = 'WCS'
+
+        # Should automatically be applied as reference to first viewer.
+        lc_plugin._obj.create_north_up_east_left(set_on_create=True)
+
+        # This would set a different reference to second viewer.
+        viewer_2 = self.imviz.create_image_viewer()
+        self.imviz.app.add_data_to_viewer("imviz-1", "has_wcs_1[SCI,1]")
+        lc_plugin.viewer = "imviz-1"
+        lc_plugin._obj.create_north_up_east_right(set_on_create=True)
+
+        assert self.viewer.state.reference_data.label == "North-up, East-left"
+        assert viewer_2.state.reference_data.label == "North-up, East-right"
+
+        # Both viewers should revert back to same reference when pixel-linked.
+        lc_plugin.link_type = 'Pixels'
+        assert self.viewer.state.reference_data.label == "has_wcs_1[SCI,1]"
+        assert viewer_2.state.reference_data.label == "has_wcs_1[SCI,1]"
 
-        lc_plugin.vue_reset_astrowidget_markers()
+        # FIXME: https://github.com/spacetelescope/jdaviz/issues/2724

```

Comment:
Note: If you can tell me what the *expected* behavior should be, I can turn this into xfail.

Reply:
I updated this test to use the *expected* answer and catch the failures for now.